aziotctl/aziotctl-common/src/status.rs

```diff
@@ -0,0 +1,125 @@
+use std::fmt;
+use std::process::Command;
+
+use crate::ServiceDefinition;
+
+#[allow(clippy::module_name_repetitions)]
+pub fn get_status(processes: &[&ServiceDefinition]) {
+    let results: Vec<ServiceStatus<'_>> = processes
+        .iter()
+        .map(|process| ServiceStatus {
+            service_name: process.service,
+            service_status: read_status(process.service),
+            sockets: process
+                .sockets
+                .iter()
+                .map(|socket| SocketStatus {
+                    socket_name: socket,
+                    socket_status: read_status(socket),
+                })
+                .collect(),
+        })
+        .collect();
+
+    if results.iter().any(|s| !s.ok()) {
+        for result in &results {
+            println!("{}: {}", result.service_name, result.service_status);
+            for socket in &result.sockets {
+                println!("{}: {}", socket.socket_name, socket.socket_status);
+            }
+            println!();
+        }
+
+        for result in &results {
+            if result.service_status == Status::Failed {
+                print_logs(result.service_name);
+            }
+            for socket in &result.sockets {
+                if !socket.ok() {
+                    print_logs(socket.socket_name);
+                }
+            }
+        }
+
+        println!("\n\nFor more detailed logs, use the `system logs` command. If the logs do not contain enough information, consider setting debug logs using `system set-log-level`.");
+    } else {
+        println!("Ok");
+    }
+}
+
+fn read_status(process: &str) -> Status {
+    let result = Command::new("systemctl")
+        .args(&["is-active", process])
+        .output()
+        .unwrap();
+
+    let output = String::from_utf8_lossy(&result.stdout);
+
+    match output.trim() {
+        "active" => Status::Active,
+        "failed" => Status::Failed,
+        "inactive" => Status::Inactive,
+        &_ => {
+            println!("\nError calling `systemctl is-active {}`.", process);
```

Comment:
There are a whole bunch of states, so I'd a) rather not treat an unexpected state as an "error calling systemctl is-active" (calling it went fine, and probably emitted valid output), and b) rather not emit a line for this that breaks up the rest of the output.

The easiest way is to make `Status::Failed` be `Status::Failed(String)` and make it hold the original string returned by `systemctl is-active`. So you silently treat it as "failed", but still have the original string to show in the status output.